@xen-orchestra/xapi/vdi.mjs

```diff
@@ -65,18 +65,25 @@ class Vdi {
   }
 
   async _connectNbdClientIfPossible(ref, { nbdConcurrency = 1 } = {}) {
-    const nbdInfos = await this.call('VDI.get_nbd_info', ref)
+    let nbdInfos = await this.call('VDI.get_nbd_info', ref)
     if (nbdInfos.length > 0) {
-      // a little bit of randomization to spread the load
-      const nbdInfo = nbdInfos[Math.floor(Math.random() * nbdInfos.length)]
+      // filter nbd to only use backup network ( if set )
+      const poolBackupNetwork = this._pool.other_config['xo:backupNetwork']
+      if (poolBackupNetwork) {
+        const networkRef = await this._roCall('network.get_by_uuid', [poolBackupNetwork])
+        const pifs = await this.getField('network', networkRef, 'PIFs')
+        // @todo implement ipv6
```

Comment:
If this is easy I think we should do it, otherwise this todo will stay for a long time :stuck_out_tongue:

Reply:
Not trivial, we'll do this in the future.